Source/MQTTnet/Packets/MqttPublishPacket.cs

```diff
@@ -2,26 +2,26 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using MQTTnet.Protocol;
 using System;
 using System.Buffers;
 using System.Collections.Generic;
-using MQTTnet.Protocol;
 
 namespace MQTTnet.Packets;
 
 public sealed class MqttPublishPacket : MqttPacketWithIdentifier
 {
     public string ContentType { get; set; }
 
-    public byte[] CorrelationData { get; set; }
+    public ReadOnlyMemory<byte> CorrelationData { get; set; }
 
     public bool Dup { get; set; }
 
     public uint MessageExpiryInterval { get; set; }
 
     public MqttPayloadFormatIndicator PayloadFormatIndicator { get; set; } = MqttPayloadFormatIndicator.Unspecified;
 
-    public ArraySegment<byte> PayloadSegment { set { Payload = new ReadOnlySequence<byte>(value); } }
+    public ReadOnlyMemory<byte> PayloadSegment { set { Payload = new ReadOnlySequence<byte>(value); } }
```

Comment:
I am wondering why we have this strange setter here. Probably a leftover from the last optimization made by @mregen?

Reply:
Yes, PayloadSegment is legacy in the low-level MqttPublishPacket, and currently almost only unit test code still uses it.